client/src/client/builder.rs

```diff
@@ -375,7 +375,7 @@ impl ClientBuilder {
             #[cfg(not(target_family = "wasm"))]
             sync_handle: sync_handle.map(Arc::new),
             #[cfg(feature = "mqtt")]
-            mqtt_client: None,
+            mqtt_client: Arc::new(tokio::sync::RwLock::new(None)),
```

Comment:
Just curious, why is it needed now ?

Reply:
Because listen() took `&mut Client` and then we would also need to make the `ClientMessageHandler` mutable for this which looked very annoying